lib/querystring.js

```diff
@@ -162,6 +162,25 @@ function stringifyPrimitive(v) {
 }
 
 
+function encodeStringified(v, encode) {
+  if (typeof v === 'string')
+    return (v.length ? encode(v) : '');
+  if (typeof v === 'number' && isFinite(v)) {
+    // Values >= 1e21 automatically switch to scientific notation which requires
+    // escaping due to the inclusion of a '+' in the output
+    return (v < 1e21 ? '' + v : encode('' + v));
```

Comment:
This needs another check to verify that `v > -1e21`

```suggestion
    return (v < 1e21 && v > -1e21 ? `${v}` : encode(`${v}`));
```